### DISCUSSION TITLE
rtcontrol purge or cull doesn't delete the data

### DISCUSSION THREAD

**Discussion by govardha** [kannibalox/pyrosimple#69] [General]
Hello,

When I do an 

```
rtcontrol hash=<hash> // --cull 
rtcontrol hash=<hash> // --purge
```
both of them acts the same way as, just the meta file is removed but the actual data isn't deleted.

`rtcontrol hash=<hash> // --delete`

**Comment by kannibalox** [comment]
I just tried it again, and it seems to work for me. The data does need to be in the sample place for both where rTorrent thinks it is, and where the rtcontrol command is actually running. I can add a debug log message about what it's trying to delete if needed, but that can quickly fill up the screen for torrents with lots of files.

**Comment by Rouneq** [comment]
Piggy-backing on this as I was just about to open a discussion on it. What I'm finding is `--cull` isn't getting rid of *all* the data. Some stray data for torrents is left as orphans. I now have a process I run after a cull to look for these orphans and issue `rm` commands for them. Comparatively, it's about 1 out of 10 or so left behind. I haven't done any sort of analysis to determine exact ratios, though. That said, I do have the data from the most recent operations from last night (I rebooted just before):

Server 1: reported 29 torrents, but three were left behind
`rtcontrol --from=data --cull --yes completed=+4d up=0 -o name,size.sz -s completed`

Server 2: reported 23 torrents, but two were left behind
`rtcontrol --from=freeleech --cull --yes completed=+4d up=0 -o name,size.sz -s completed`

I waited roughly 10 minutes or so before checking orphan state on the servers.

I think adding either a new command-line option, e.g. `--cull-verbose`, or enhancing the existing `--debug` command-line option to kick out what actions are happening with `--cull` would be a good start.
